server/server.go

```diff
@@ -113,24 +113,24 @@ func (s *Server) Version(ctx context.Context, req *v1alpha1.VersionRequest) (*v1
 func handleMountEvent(ctx context.Context, client *secretmanager.Client, cfg *config.MountConfig, writeSecrets bool) (*v1alpha1.MountResponse, error) {
 	results := make([]*secretmanagerpb.AccessSecretVersionResponse, len(cfg.Secrets))
 	errs := make([]error, len(cfg.Secrets))
+	g, ctx := errgroup.WithContext(ctx)
 
 	// In parallel fetch all secrets needed for the mount
-	wg := sync.WaitGroup{}
 	for i, secret := range cfg.Secrets {
-		wg.Add(1)
-
-		i, secret := i, secret
-		go func() {
-			defer wg.Done()
+		i, secret := i, secret // https://golang.org/doc/faq#closures_and_goroutines
+		g.Go(func() error {
 			req := &secretmanagerpb.AccessSecretVersionRequest{
 				Name: secret.ResourceName,
 			}
 			resp, err := client.AccessSecretVersion(ctx, req)
 			results[i] = resp
 			errs[i] = err
-		}()
+			return nil
+		})
+	}
+	if err := g.Wait(); err != nil {
```

Comment:
errgroup is not used because it will only return the first error, instead errors are collected into `errs` so that all of them can be reported with the `buildErr` helper

Reply:
Which is does in this case, right? `g.Wait()` will return error if any of the `g.Go` functions returned an error. In this case all of them return nil, and the errors are collected in the same way as before. So I think this is semantically the same, but I could be wrong.

Reply:
No, if `errgroup` is used, line 132 will cause `handleMountEvent` to return with just 1 error, as the grouping isnt done until  line 144

Reply:
ohhh i see you have it return nil on line 128 to make the tests still pass. I'm not convinced that using errgroup provides any benefits in this structure though?